### ISSUE TITLE
NordVPN issue

### ISSUE THREAD

**Issue by Mameluk007** [haugene/docker-transmission-openvpn#953]
Hi everyone !

First, thanks a lot for this rep which is exactly what I was looking for. 

Though it doesn't work as expected : it worked once (yesterday when I ran it for the first time) but now it doesn't work anymore despite no change made. It looks like the NORDVPN_xxx ENV variables are not passed to the script dedicated to find a suitable server (updateConfigs.sh) resulting in an error. 

I run docker on a Synology NAS with a Nordvpn. 

Below are this log and the ENV variables regarding NORDVPN : 

2019-11-17 21:58:29 | stdout | Using OpenVPN provider: NORDVPN
2019-11-17 21:58:29 | stdout | 2019-11-17 22:58:29 Checking curl installation
 2019-11-17 21:58:29 | stdout | 2019-11-17 22:58:29 Removing existing configs
2019-11-17 21:58:29 | stdout | 2019-11-17 22:58:29 Selecting the best server...
2019-11-17 21:58:49 | stdout | 2019-11-17 22:58:29 Searching for technology: openvpn_udp
2019-11-17 22:58:29 Unable to find a server with the specified parameters, using any recommended server
2019-11-17 21:59:30 | stdout | 2019-11-17 22:58:29 Best server :
2019-11-17 21:59:30 | stdout | 2019-11-17 22:58:29 Downloading config: default.ovpn
2019-11-17 22:58:29 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/.udp.ovpn
...

ENV variables 
NORDVPN_COUNTRY = FR
NORDVPN_PROTOCOL = UDP
...

Thanks for your help.

Stef

**Comment by Raka74** [comment]
Hi!

I noticed the same thing.
Nothing is downloaded from the url in the updateConfigs script and since all existing are cleared there’s no connection possible after that it seems.

For now I just removed the script from the nordvpn folder and I’m using the default without fallback.

**Comment by Mameluk007** [comment]
Thanks for your feedback ! Do you use an API (like I do on a Syno NAS) or do you run docker with a line of command ? 
Where is the default ovpn file supposed to be placed ? 
Thanks.

**Comment by Raka74** [comment]
I’m running Docker on a Synology as well but when I didn’t get further using the UI alone I created the container in ssh like so:

sudo docker run --cap-add=NET_ADMIN --device=/dev/net/tun -d \
              -v /volume1/docker/T2/:/data \
              -v /etc/localtime:/etc/localtime:ro \
              --env-file /volume1/docker/temp/DockerEnv.txt \
              -p 9091:9091 \
              haugene/transmission-openvpn

In the DockerEnv.txt is all the needed info like provider /. Password and such

https://github.com/haugene/docker-transmission-openvpn/blob/master/DockerEnv


When done I see it running in the UI.

I then again in ssh I listed the ID of the container with 

docker ps -a

Then entered it with 

docker exec -it [container-id] bash

Navigated to /etc/openvpn/nordvpn and there removed the updateconfigs script

When done I went back in to the graphic UI and setup the rest to my liking + added a resolv.conf

![3DB0519C-7E86-4CAD-8AE6-357CA931D0E6](https://user-images.githubusercontent.com/40692094/69083657-d8d24680-0a42-11ea-8e76-9f65f751404b.jpeg)

**Comment by edgd1er** [comment]
As the log showed, no server was found: 
```
2019-11-17 21:59:30 | stdout | 2019-11-17 22:58:29 Best server :
```
It could be an api error or an env setting typo.

I've been running haugene openvpn with NORDVPN for a while now. Yesterday, I had to reinstall the container. besides password and username, three env are expected by the script:

* NORDVPN_COUNTRY: (values code: https://api.nordvpn.com/v1/servers/countries)
* NORDVPN_CATEGORY: (values identifier: https://api.nordvpn.com/v1/servers/groups)
* NORDVPN_PROTOCOL: tcp or udp (values identifier more available at https://api.nordvpn.com/v1/technologies, may need script adaptation)

more info if needed in https://github.com/haugene/docker-transmission-openvpn/blob/master/docs/nordvpn-script.md

**Comment by Raka74** [comment]
Aha that’s new to me, thanks edgd1er.

I din’t know about these env variables needed. 
Maybe when I updated to the latest version and imported my saved settings it didn’t work because they were missing.

I will check if this works.

**Comment by timothymiller** [comment]
I encountered this issue yesterday. There are a couple things you must modify if you want to use NordVPN with this docker container.
**Add** the following environment variable:
`NORDVPN_COUNTRY=<your_country_code_here>`
**Delete** the following environment variable:
`OPENVPN_CONFIG=default`

**Comment by Raka74** [comment]
Yes,

After edgd1er ‘s post I filled out the variables as mentioned here:

https://github.com/haugene/docker-transmission-openvpn/blob/master/docs/nordvpn-script.md
Or 
https://haugene.github.io/docker-transmission-openvpn/nordvpn-script/

And all is running fine again.
I did not know that these were mandatory for a config with NordVPN to work. Just a case of rtfm I guess.

As far as I’m concerned this is solved. But the person that opened this issue hasn’t commented back.

**Comment by haugene** [comment]
Thanks for the update @Raka74. As you say, it's in the manuals, but I see that this is a recurring issue so it could have been made clearer. I'm thinking of restructuring the github pages somewhat. Or at least making a better overview so that people find their relevant sections.

I'm closing this for now. @Mameluk007, feel free to re-open if it still doesn't work for you.

**Comment by Mameluk007** [comment]
Hi !

Thank you very much all of you for your feedback. I am not able to test this right now but when I am I’ll let you know.

See you

**Comment by Mameluk007** [comment]
Hi everybody !

Each of the issue encountered previously have bee solved. Though I have another one : I have shitty bandwidth in comparison with my previous config (transmission run directly as a regular app under my Synology NAS through a VPN). Obvisouly I ran bandwidth tests with exactly the same VPN servers with the same protocol and results are far slower !
I checked if any firewall or bandwith limitation was active but there is not... 

Anybody encoutered the same issue ? 

Thanks.
